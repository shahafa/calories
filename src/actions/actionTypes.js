export const RESET_AUTH_STORE = 'RESET_AUTH_STORE';
export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const GET_MEALS_REQUEST = 'GET_MEALS_REQUEST';
export const GET_MEALS_SUCCESS = 'GET_MEALS_SUCCESS';
export const GET_MEALS_FAILURE = 'GET_MEALS_FAILURE';
export const ADD_MEAL_REQUEST = 'ADD_MEAL_REQUEST';
export const ADD_MEAL_SUCCESS = 'ADD_MEAL_SUCCESS';
export const ADD_MEAL_FAILURE = 'ADD_MEAL_FAILURE';
export const DELETE_MEAL_REQUEST = 'DELETE_MEAL_REQUEST';
export const DELETE_MEAL_SUCCESS = 'DELETE_MEAL_SUCCESS';
export const DELETE_MEAL_FAILURE = 'DELETE_MEAL_FAILURE';
export const EDIT_MEAL_REQUEST = 'EDIT_MEAL_REQUEST';
export const EDIT_MEAL_SUCCESS = 'EDIT_MEAL_SUCCESS';
export const EDIT_MEAL_FAILURE = 'EDIT_MEAL_FAILURE';
export const CLOSE_MEALS_ERROR_SNACKBAR = 'CLOSE_MEALS_ERROR_SNACKBAR';
export const SET_FILTER = 'SET_FILTER';

export const GET_USERS_REQUEST = 'GET_USERS_REQUEST';
export const GET_USERS_SUCCESS = 'GET_USERS_SUCCESS';
export const GET_USERS_FAILURE = 'GET_USERS_FAILURE';
export const UPDATE_USERS_ROLE_REQUEST = 'UPDATE_USERS_ROLE_REQUEST';
export const UPDATE_USERS_ROLE_SUCCESS = 'UPDATE_USERS_ROLE_SUCCESS';
export const UPDATE_USERS_ROLE_FAILURE = 'UPDATE_USERS_ROLE_FAILURE';

export const GET_SETTINGS_REQUEST = 'GET_SETTINGS_REQUEST';
export const GET_SETTINGS_SUCCESS = 'GET_SETTINGS_SUCCESS';
export const GET_SETTINGS_FAILURE = 'GET_SETTINGS_FAILURE';
export const SET_SETTINGS_REQUEST = 'SET_SETTINGS_REQUEST';
export const SET_SETTINGS_SUCCESS = 'SET_SETTINGS_SUCCESS';
export const SET_SETTINGS_FAILURE = 'SET_SETTINGS_FAILURE';
export const CLOSE_SETTINGS_ERROR_SNACKBAR = 'CLOSE_SETTINGS_ERROR_SNACKBAR';
