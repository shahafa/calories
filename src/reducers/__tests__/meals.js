import mealsReducer from '../meals';
import * as types from '../../constants/actionTypes';

describe('meals reducer', () => {
  it('should return the initial state', () => {
    expect(
      mealsReducer(undefined, {}),
    ).toMatchSnapshot();
  });

  it('should start meals request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.GET_MEALS_REQUEST,
      }),
    ).toMatchSnapshot();
  });

  it('should set meals after successful GET meals request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.GET_MEALS_SUCCESS,
        payload: {
          data: {
            meals: [{
              id: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
              date: '2017-06-14T11:04:00Z',
              userEmail: 'user@mail.com',
              meal: 'test meal',
              calories: 100,              
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should set meals loading flag to false after GET meals failure', () => {
    expect(
      mealsReducer(undefined, {
        type: types.GET_MEALS_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should add meal locally before add meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.ADD_MEAL_REQUEST,
        meta: {
          meal: {
            id: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
            date: '2017-06-14T11:04:00Z',
            userEmail: 'user@mail.com',
            meal: 'test meal',
            calories: 100,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should update meals after successful add meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.ADD_MEAL_SUCCESS,
        payload: {
          data: {
            meals: [{
              id: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
              date: '2017-06-14T11:04:00Z',
              userEmail: 'user@mail.com',
              meal: 'test meal',
              calories: 100,
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert meals list after add meal failure', () => {
    expect(
      mealsReducer(undefined, {
        type: types.ADD_MEAL_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should delete meal locally before delete meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.DELETE_MEAL_REQUEST,
        meta: {
          mealId: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
        },
      }),
    ).toMatchSnapshot();
  });

  it('should update meals after successful delete meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.DELETE_MEAL_SUCCESS,
        payload: {
          data: {
            meals: [],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert meals list after delete meal failure', () => {
    expect(
      mealsReducer(undefined, {
        type: types.DELETE_MEAL_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should update meal locally before edit meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.EDIT_MEAL_REQUEST,
        meta: {
          meal: {
            id: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
            date: '2017-06-14T11:04:00Z',
            userEmail: 'user@mail.com',
            meal: 'test meal',
            calories: 105,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should update meals after successful edit meal request', () => {
    expect(
      mealsReducer(undefined, {
        type: types.EDIT_MEAL_SUCCESS,
        payload: {
          data: {
            meals: [{
              id: '7eacca01-5a0e-4956-badc-a60e98eb3cfe',
              date: '2017-06-14T11:04:00Z',
              userEmail: 'user@mail.com',
              meal: 'test meal',
              calories: 105,
            }],
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should revert meals list after edit meal failure', () => {
    expect(
      mealsReducer(undefined, {
        type: types.EDIT_MEAL_FAILURE,
      }),
    ).toMatchSnapshot();
  });

  it('should set filter', () => {
    expect(
      mealsReducer(undefined, {
        type: types.SET_FILTER,
        filter: {
          fromDate: '2017-06-11T21:00:00.000Z',
          fromTime: '2017-06-14T11:44:18.187Z',
          toDate: '2017-06-14T11:44:19.930Z',
          toTime: '2017-06-14T11:44:23.220Z',
        },
      }),
    ).toMatchSnapshot();
  });
});
