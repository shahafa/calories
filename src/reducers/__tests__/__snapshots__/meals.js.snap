// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meals reducer should add meal locally before add meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 100,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;

exports[`meals reducer should delete meal locally before delete meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should return the initial state 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should revert meals list after add meal failure 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 100,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;

exports[`meals reducer should revert meals list after delete meal failure 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should revert meals list after edit meal failure 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 105,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;

exports[`meals reducer should set filter 1`] = `
Object {
  "filter": Object {
    "fromDate": "2017-06-11T21:00:00.000Z",
    "fromTime": "2017-06-14T11:44:18.187Z",
    "toDate": "2017-06-14T11:44:19.930Z",
    "toTime": "2017-06-14T11:44:23.220Z",
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should set meals after successful GET meals request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 100,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;

exports[`meals reducer should set meals loading flag to false after GET meals failure 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should start meals request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": true,
  "meals": Array [],
}
`;

exports[`meals reducer should update meal locally before edit meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should update meals after successful add meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 100,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;

exports[`meals reducer should update meals after successful delete meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [],
}
`;

exports[`meals reducer should update meals after successful edit meal request 1`] = `
Object {
  "filter": Object {
    "fromDate": null,
    "fromTime": null,
    "showAll": false,
    "toDate": null,
    "toTime": null,
  },
  "isLoading": false,
  "meals": Array [
    Object {
      "calories": 105,
      "date": "2017-06-14T11:04:00Z",
      "id": "7eacca01-5a0e-4956-badc-a60e98eb3cfe",
      "meal": "test meal",
      "userEmail": "user@mail.com",
    },
  ],
}
`;
